hydra/core/override_parser/types.py

```diff
@@ -258,7 +268,10 @@ def _convert_value(value: ParsedElementType) -> Optional[ElementType]:
         if isinstance(value, list):
             return [Override._convert_value(x) for x in value]
         elif isinstance(value, dict):
-            return {k: Override._convert_value(v) for k, v in value.items()}
+            return {
```

Comment:
Do you have any test covering this change?
I am not sure I understand why it's needed or what the implications are to callers.

Reply:
Both tests I added in `test_hydra.py` fail without this change.

The error looks like this:
```
Error merging override foo={'quoted_$(){}[]': 1, id123: 1, 123id: 1, a/-\+.$%*@: 1, \\\(\)\[\]\{\}\:\=\ \       \,: 1, white space: 1}
Incompatible key type 'QuotedString'
        full_key:
        reference_type=Optional[Dict[Union[str, Enum], Any]]
        object_type=dict
```

What happens is that otherwise, quoted strings remain in the dictionary passed to OmegaConf, which doesn't like them.

Reply:
Can you create a lower level test for this?

Reply:
> Can you create a lower level test for this?

So, currently the lower level test is the one I moved in 6bcaac207 (`test_dict_key_formats()`, is it lower level enough?

Reply:
> Still feels misplaced. After adding the lower level tests in the grammar parser, do you still thank that test_dict_key_formats() is needed?

As discussed elsewhere, the new test that covers the change on that line is the one from 2e545f9ec